src/main/java/dk/kb/util/yaml/YAML.java

```diff
@@ -906,23 +935,29 @@ private void conditionalTraverse(List<String> yPath, MultipleValuesVisitor visit
      * @param yPath list of path elements, where the first element is the current element of the YAML traversal, which
      *              is checked for '[' and ']'.
      */
-    private static void removeBracketsFromPathElement(List<String> yPath) {
-        if (!yPath.isEmpty() && yPath.get(0).startsWith("[") && yPath.get(0).endsWith("]")){
-            String cleaned = yPath.get(0).substring(1, yPath.get(0).length()-1);
-            yPath.set(0, cleaned);
+    private static YPath removeBracketsFromPathElement(YPath yPath) {
+        Matcher arrayMatcher = ARRAY_ELEMENT.matcher(yPath.getFirst());
+
+        if (arrayMatcher.matches()){
+            String cleaned = arrayMatcher.group(2);
+            YPath replacedYPath = yPath.replaceFirst(cleaned);
+
+            return replacedYPath;
+        } else {
```

Comment:
Very minor: Ad the `if` ends with a `return`, there is no need for wrapping `return yPath` in an `else`.